internal/controllers/core/cluster/reconciler.go

```diff
@@ -50,6 +53,11 @@ func NewReconciler(ctrlClient ctrlclient.Client, store store.RStore, localDocker
 	}
 }
 
+func (r *Reconciler) SetFakeClientsForTesting(k8sClient k8s.Client, dockerClient docker.Client) {
+	r.fakeK8sClient = k8sClient
+	r.fakeDockerClient = dockerClient
+}
```

Comment:
No objection to this right now since it's not particularly invasive, but do you think we should move to having factory interfaces for the clients eventually?

Reply:
i kind of suspect that we'll inject a fake client cache (as in https://github.com/tilt-dev/tilt/pull/5284) and use that everywhere?

Reply:
(assuming that the client cache is more like a client factory)